test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -503,6 +503,22 @@
       (is (true? (revoked? cert1)))
       (is (true? (revoked? cert2))))))
 
+(deftest filtering-serials-to-be-add-to-revoked-list
+  (testing "The filtering function return an empty set"
+    (let [crl (-> (get-certificate-revocation-list cacrl)
+                  StringReader.
+                  utils/pem->crl)
+          test-serial (vec [4])]
+      (is (empty? (filter-already-revoked-serials test-serial crl)))))
+
+  (testing "The filtering function return a set of serials not yet in CRL"
+    (let [crl (-> (get-certificate-revocation-list cacrl)
```

Comment:
since you're using the same CRL for both of these tests, you could put this in a `let` just inside the `deftest`, before each of the `testing` blocks.

Reply:
Definitely! I'm looking into whether I should add tests related to revoking but it seems that the existing tests already cover most of the scenario.